tsdb/wlog/wlog.go

```diff
@@ -610,6 +616,11 @@ func (w *WL) pagesPerSegment() int {
 func (w *WL) Log(recs ...[]byte) error {
 	w.mtx.Lock()
 	defer w.mtx.Unlock()
+	defer func() {
+		if w.WriteNotified != nil {
+			w.WriteNotified.Notify()
```

Comment:
How much notify is too much notify? Alternatively, we can call this once at the end of logging in a TSDB commit ([here](https://github.com/prometheus/prometheus/blob/main/tsdb/head_append.go#L844)) because a single commit (hence the scrape) can have more than one WAL logging (more common once native histograms get adopted).

Reply:
I'll move the call to Notify, good call.

Reply:
I'm not 100% sure if the only creation of the `head` struct for a normal prometheus (not read-only) is via the db `open` call? Is there a similar `FlushWAL` that could create another new `head` struct? If not I can clean up the saving of the reference to the interface even further than my most recent commit.

Reply:
bump @codesome